### PR TITLE
refactor(planarControls): refactor planar controls as an ES6 class

### PR DESCRIPTION
**Pull request by mgermerie**
## Description

Refactor PlanarControls implementation (in `src/Controls/PlanarControls.js`) as an ES6 class (it was previously a method).

### CODE REVIEW

File: src/Controls/PlanarControls.js [hunk 1]
```diff
@@ -15,6 +15,13 @@ const mouseButtons = {
     RIGHTCLICK: THREE.MOUSE.RIGHT,
 };
 
+// starting camera position and orientation target
+const startPosition = new THREE.Vector3();
```
Comment (gchoqueux) [inline src/Controls/PlanarControls.js:19]: Good idea to instantiate only once

File: src/Controls/PlanarControls.js [hunk 2]
```diff
@@ -26,197 +33,230 @@ const STATE = {
...
+const travelStartPos = new THREE.Vector3();
+const travelStartRot = new THREE.Quaternion();
+const travelEndRot = new THREE.Quaternion();
+let travelAlpha = 0;
+let travelDuration = 0;
+let travelUseRotation = false;
+let travelUseSmooth = false;
+
+// default parameters :
+const defaultEnableRotation = true;
```
Comment (gchoqueux) [inline src/Controls/PlanarControls.js]: create `default` object, it's more readable

```js
const default  = {
EnableRotation: true,
RotateSpeed: 2.0,
MinPanSpeed: 0.05,
MaxPanSpeed: 15,
ZoomTravelTime: 0.2,
ZoomInFactor: 0.25,
ZoomOutFactor: 0.4,
MaxAltitude: 12000,
GroundLevel: 200,
//...
};

File: src/Controls/PlanarControls.js [hunk 3]
```diff
@@ -26,197 +33,230 @@ const STATE = {
...
+const defaultSmartZoomHeightMax = 500;
+const defaultInstantTravel = false;
+const defaultMinZenithAngle = 0;
+const defaultMaxZenithAngle = 82.5;
+const defaultFocusOnMouseOver = true;
+const defaultFocusOnMouseClick = true;
+const defaultHandleCollision = true;
+const defaultMinDistanceCollision = 30;
+
+export const PLANAR_CONTROL_EVENT = {
```
Comment (gchoqueux) [inline src/Controls/PlanarControls.js]: It's another PR?

File: src/Controls/PlanarControls.js [hunk 4]
```diff
@@ -26,197 +33,230 @@ const STATE = {
...
+                this.minDistanceCollision,
+            );
+        }
+        // dt will not be relevant when we just started rendering. We consider a 1-frame move in this case
         if (updateLoopRestarted) {
             dt = 16;
         }
+        if (STATE.NONE !== this.state) {
+            this.view.dispatchEvent({ type: PLANAR_CONTROL_EVENT.MOVED });
+        }
```
Comment (gchoqueux) [inline src/Controls/PlanarControls.js]: It's another PR?

File: src/Controls/PlanarControls.js [hunk 5]
```diff
@@ -234,183 +274,177 @@ function PlanarControls(view, options = {}) {
...
+            0,
+            1,
+        );
+
+        // pan movement speed, adjusted according to altitude
+        const panSpeed = THREE.MathUtils.lerp(
+            this.minPanSpeed,
+            this.maxPanSpeed,
+            distToGround,
+        );
```
Comment (gchoqueux) [inline src/Controls/PlanarControls.js:344]: In next PR, could you avoid to change the formatting, it adds diffs and is harder to review

File: src/Controls/PlanarControls.js [hunk 6]
```diff
@@ -234,183 +274,177 @@ function PlanarControls(view, options = {}) {
...
+     * is previously initiated in [initiateRotation]{@link PlanarControls#initiateRotation}.
      * Compute the new position value and update the camera controls.
      *
-     * @function
      * @ignore
      */
-    this.handleRotation = (() => {
-        const vec = new THREE.Vector3();
+    handleRotation() {
+        const vect = new THREE.Vector3();
```
Comment (gchoqueux) [inline src/Controls/PlanarControls.js]: could instance `vect` and `quad` before Class? (or use free variable already instanced)

File: src/Controls/PlanarControls.js [hunk 7]
```diff
@@ -234,183 +274,177 @@ function PlanarControls(view, options = {}) {
...
+     * @param   {Event} event   the mouse wheel event.
      * @ignore
      */
-    this.initiateZoom = function initiateZoom(event) {
+    initiateZoom(event) {
         let delta;
 
         // mousewheel delta
-        if (event.wheelDelta !== undefined) {
+        if (undefined !== event.wheelDelta) {
```
Comment (gchoqueux) [inline src/Controls/PlanarControls.js:439]: could you minimize the differences?

File: src/Controls/PlanarControls.js (hunk 2 above)
Comment (mgermerie) [reply src/Controls/PlanarControls.js]: did it by creating a `defaultOptions` object (the `default` name is already used as a keyword).

**Review by gchoqueux — APPROVED** [review]

**Pull request merged**